### PR TITLE
Configurable scores with exponential decay

### PR DESCRIPTION
**Pull request by njoyard**
This PR implements exponential decay as shown in #63.

It also introduces a new Setting model to store administration preferences. For now they only contain score decay parameters ; default value are chosen so that score calculation stays the same as what we currently have.

There are no tests in the memopol_settings app because there is no business logic, only a model and an admin view. It is a separate app because it will most likely be used by the memopol app in the future (it has no reason to be in representatives_recommendations, but it can't be in memopol because then there would be a circular dependency between representatives_recommendations and memopol).

### CODE REVIEW

**Comment by jpic** [comment]
Did you consider using [django-constance](https://django-constance.readthedocs.io/en/latest/) for settings ? I like that it can type-check settings.

**Comment by njoyard** [comment]
@jpic yes, but it uses pickle. That makes it a pain to retrieve the value from SQL.

File: representatives_recommendations/migrations/0006_score_formula.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                    * (CASE
+                        WHEN representatives_votes_vote."position"::text = representatives_recommendations_recommendation.recommendation::text THEN representatives_recommendations_recommendation.weight
+                        ELSE 0 - representatives_recommendations_recommendation.weight
+                    END)) AS INTEGER) AS score
+            FROM representatives_votes_vote
+                JOIN (SELECT CAST(TO_NUMBER(value, '99999') AS FLOAT) AS value FROM memopol_settings_setting WHERE key = 'SCORE_DECAY_NUM') decay_num ON 1=1
+                JOIN (SELECT CAST(TO_NUMBER(value, '99999') AS FLOAT) AS value FROM memopol_settings_setting WHERE key = 'SCORE_DECAY_DENOM') decay_denom ON 1=1
+                JOIN (SELECT CAST(TO_NUMBER(value, '99999') AS FLOAT) AS value FROM memopol_settings_setting WHERE key = 'SCORE_EXPONENT') exponent ON 1=1
+                JOIN representatives_votes_proposal ON representatives_votes_vote.proposal_id = representatives_votes_proposal.id
+                LEFT JOIN representatives_recommendations_recommendation ON representatives_votes_proposal.id = representatives_recommendations_recommendation.proposal_id
```
Comment (jpic) [inline representatives_recommendations/migrations/0006_score_formula.py]: :guitar:

**Comment by jpic** [comment]
I see, pretty nasty, I wonder why they did that. LGTM :guitar:

**Comment by njoyard** [comment]
They did that to have a cool multi-type column. That's great, but that's not our need here. We need a quick way of having settings reachable for SQL ;)

**Comment by jpic** [comment]
Interresting, thanks for answering, so, are you sure we really want a charfield here ?

**Comment by njoyard** [comment]
As this is a generic settings table, yes, I think a charfield is the best. Even if for now, we only have numeric values. But the cast has a negligible cost (only a few lines cast to numbers by PG when querying the score views)

**Comment by jpic** [comment]
I see, reusability is good to have, lgtm ;)

**Comment by njoyard** [comment]
Rewrote things :)
Not a huge change but all score fields became FloatFields, there is an additional SCORE_DECIMALS parameters and the score rounding is now customizable (defaults to integers only).

**Pull request merged**